portfolio/src/main/webapp/apiLibrary.js

```diff
@@ -0,0 +1,168 @@
+  } else {
+    logOutAllApis();
+  }
+}
+
+/**
+ * Updates import message box based on the error during authentication process
+ * for importing.
+ */
+function handleImportAuthError(e) {
+  var $importAuthMessage = $('#import-auth-message');
+
+  var errorMessage;
+  if (e.error === ERROR_CODES.popup_closed_by_user) {
+    errorMessage =
+        'It seems like you didn\'t complete the authorization process. ' +
+        'Please click the Login button again.'
+  } else if (e.error === ERROR_CODES.access_denied) {
+    errorMessage =
+        'You didn\'t give permission to view your Google Account, ' +
+        'so your information cannot be viewed or imported.'
+  } else {
+    errorMessage = 'An error occurred.';
+  }
+  $importAuthMessage.text(errorMessage).removeClass('d-none');
+}
+
+/**
+ * If current use is signed in and authorized,
+ * for read-only access to any of the APIs, 
+ * hide the log-in button and show log-out button for that API.
+ * Otherwise, the user needs to log in and/or authorize. 
+ */
+function handleApiButtons() {
+  const $logInButton = $('#google-auth-button');
+  const $logOutButton = $('#google-logout-button');
+  const $exportSheetsButton = $('#sheets-export-button'); 
+  const $connectTasksButton = $('#connect-tasks-btn'); 
+  const $importCalendarButton = $('#import-calendar-button'); 
+
+  if (GoogleAuth && GoogleAuth.isSignedIn.get()) {
+    currentUser = GoogleAuth.currentUser.get(); 
+    if (currentUser.hasGrantedScopes(SCOPE_CALENDAR_READ_ONLY) ||
+        currentUser.hasGrantedScopes(SCOPE_CALENDAR_READ_WRITE)){
+      $logInButton.addClass('d-none');
+      $logOutButton.removeClass('d-none');
+      $importCalendarButton .removeClass('d-none');
+    } else {
+      $logInButton.removeClass('d-none');
+      $logOutButton.addClass('d-none'); 
+      $importCalendarButton .addClass('d-none');
+    }
+
+    if (currentUser.hasGrantedScopes(SCOPE_TASKS_READ_ONLY)) {
+      $connectTasksButton.removeClass('d-none');
+    } else {
+      $connectTasksButton.addClass('d-none');
+    }
+
+  } 
+}
+
+/**
+ * Handles signing out of the user's Google account.
+ * Hides all API buttons accordinly, except the main "Log into your Google account" button.
+ */
+function logOutAllApis() {
+  const $calendarView = $('#calendar-view');
+  const $logInButton = $('#google-auth-button');
+  const $logOutButton = $('#google-logout-button');
+  const $exportSheetsButton = $('#sheets-export-button'); 
+  const $connectTasksButton = $('#connect-tasks-btn'); 
+  const $importCalendarButton = $('#import-calendar-button');
+  const $exportCalendarButton = $('#export-calendar-button'); 
+
+  $calendarView.addClass('d-none');
+  $logInButton.removeClass('d-none');
+  $logOutButton.addClass('d-none');
+  $exportCalendarButton.addClass('d-none'); 
+  $exportSheetsButton.addClass('d-none');
+  $connectTasksButton.addClass('d-none'); 
+  $importCalendarButton.addClass('d-none'); 
+
+  clearImportMenu();
+
+  GoogleAuth.signOut();
+}
```

Comment:
nit: I think a lot of the files need an extra new line at the end.

Reply:
Done.